### ISSUE TITLE
Flags series allowOverlapX not working when flags in diffent series

### ISSUE THREAD

**Issue by slavede** [highcharts/highcharts#21263]
#### Expected behaviour
I expect algorithm to be applied on all flags so they don't overlap (or provide someway of clustering)

#### Actual behaviour
Overlap is happening

#### Live demo with steps to reproduce

https://jsfiddle.net/v439uqmw/
https://jsfiddle.net/xp2v9m0c/1/

#### Product version
11

#### Affected browser(s)
Chrome

**Comment by karolkolodziej** [comment]
Hi @slavede! 

The `allowOverlapX` property was designed to work only within the same series.
If you have multiple series, the algorithm is unaware of that and can not reposition them.

[See the demo comparison between the two.](https://jsfiddle.net/BlackLabel/3L5b21wo/)

I would not consider this as a bug.

**Comment by slavede** [comment]
I would agree then it's not a bug. Maybe we can change it to feature request?

**Comment by slavede** [comment]
I have actually opened this because I can't use squarepin and color every pin differently (add className to a point). I can add className per flag series. That's why I need multiple series (to color points differently).

If I would be able to add className to point, this would be a less of an issue because they would live in same series. Would you consider accepting this as a new feature request? (I can open a ticket)

**Comment by karolkolodziej** [comment]
Isn't it how `stacking` should work @pawelfus? 
[It seems to work when all flags are in one series](https://jsfiddle.net/gh/get/library/pure/highcharts/highcharts/tree/master/samples/stock/plotoptions/flags-stackdistance/), but as soon as you have more of them it breaks and they overlap.

**Comment by pawelfus** [comment]
> I have actually opened this because I can't use squarepin and color every pin differently (add className to a point). I can add className per flag series. That's why I need multiple series (to color points differently).

The question is wether you want to set color per flag, or set className per flag? 

Color is supported, demo: https://jsfiddle.net/BlackLabel/41kw0o79/ (see A and B flags)

ClassName is not, which is a bug, and should be reported separately: https://jsfiddle.net/BlackLabel/41kw0o79/ (see C flag) @karolkolodziej 

Worakround for the Classname: https://jsfiddle.net/BlackLabel/41kw0o79/1/ 

*Internal note:*
We should try using `getClassName()` instead of hardcoded `highcharts-point` here: https://github.com/highcharts/highcharts/blob/93424115d125a9eeb2e0985182b467fe9afccf17/ts/Series/Flags/FlagsSeries.ts#L214 Unless there is a test which will block us, then consider just adding the `options.className`

**Comment by slavede** [comment]
Hi @pawelfus , thanks for the answer (and workaround).

Color is supported only in case you are not using styled mode, in styled mode color is ignored and then I would like to use className to apply color based on class.

I agree that missing support for class per point would be a bug.

Should I open a new ticket or it can be continued on this ticket?

**Comment by karolkolodziej** [comment]
Let's keep this ticket open. We will investigate why the API documentation is different for the Highcharts and Highcharts Stock see:
![image](https://github.com/highcharts/highcharts/assets/57004120/f4b266f3-f54f-41fb-8918-9f2dd5e4de01)
![image](https://github.com/highcharts/highcharts/assets/57004120/28b949d5-4cc7-422c-b637-56874d51efb8)
